react/features/toolbar/components/ToolbarButton.web.js

```diff
@@ -0,0 +1,190 @@
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props: Object) {
+        super(props);
+
+        this.popups = [];
+
+        // Bind methods to save the context
+        const self: any = this;
+
+        self._pushPopup = this._pushPopup.bind(this);
+
+    }
+
+    /**
+     * Forces translation of the button and sets shortcut/tooltip
+     * after mounting of the component.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        this.popups.forEach(popup => {
+            APP.translation.translateElement($(popup));
+        });
+
+        this._setShortcutAndTooltip();
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        type MapOfProps = { [key: string]: * };
+
+        let className = this.props.className;
+
+        if (this.props.toggled) {
+            className += ' toggled';
+        }
+
+        const popups = this.props.popups || [];
+        const id = this.props.id;
+        const props: MapOfProps = {
+            className,
+            id
+        };
+
+        props['data-container'] = 'body';
+        props['data-placement'] = 'bottom';
+
+        props.onClick = this.props.onClick;
+
+        if (this.props.content) {
+            props.content = this.props.content;
+        }
+
+        if (this.props.i18n) {
+            props['data-i18n'] = this.props.i18n;
+        }
+
+        // TODO: remove it after UI.updateDTMFSupport fix
+        if (this.props.hidden) {
+            props.style = { display: 'none' };
+        }
+
+        return (
+            <a { ...props } >
+                { this._renderInnerElementsIfRequired() }
+                { this._renderPopups(popups) }
+            </a>
+        );
+    }
+
+    /**
+     * If toolbar button should contain children elements
+     * renders them.
+     *
+     * @returns {ReactElement|null}
+     * @private
+     */
+    _renderInnerElementsIfRequired() {
+        if (this.props.html) {
+            return this.props.html;
+        }
+
+        return null;
+    }
+
+    /**
+     * Renders popup element for toolbar button.
+     *
+     * @param {Array} popups - Array of popup objects.
+     * @returns {Array}
+     * @private
+     */
+    _renderPopups(popups: Array<*> = []) {
+        return popups.map(popup => {
+            let gravity = 'n';
+
+            if (popup.dataAttrPosition) {
+                gravity = popup.dataAttrPosition;
+            }
+
+            return (
+                <div
+                    className = { popup.className }
+                    data-i18n = { popup.dataAttr }
+                    data-tooltip = { gravity }
+                    id = { popup.id }
+                    key = { popup.id }
+                    ref = { this._pushPopup } />
```

Comment:
we use popups array only for translations. Since I've added translation of toolbar button after every update we don't need this array at all.